src/Servers/IIS/AspNetCoreModuleV2/AspNetCore/ShimOptions.cpp

```diff
@@ -82,4 +84,33 @@ ShimOptions::ShimOptions(const ConfigurationSource &configurationSource) :
     auto dotnetEnvironmentEnabled = equals_ignore_case(L"Development", dotnetEnvironment);
 
     m_fShowDetailedErrors = detailedErrorsEnabled || aspnetCoreEnvironmentEnabled || dotnetEnvironmentEnabled;
+
+    // Specifies how long to delay (in milliseconds) after IIS tells us to stop before starting the application shutdown.
+    // See StartShutdown in globalmodule to see how it's used.
+    auto shutdownDelay = find_element(handlerSettings, CS_ASPNETCORE_SHUTDOWN_DELAY).value_or(std::wstring());
+    if (shutdownDelay.empty())
+    {
+        // Fallback to environment variable if process specific config wasn't set
+        shutdownDelay = Environment::GetEnvironmentVariableValue(CS_ASPNEtCORE_SHUTDOWN_DELAY_ENV)
+            .value_or(environmentVariables[CS_ASPNEtCORE_SHUTDOWN_DELAY_ENV]);
+        if (shutdownDelay.empty())
+        {
+            // Default if neither process specific config or environment variable aren't set
+            m_fShutdownDelay = std::chrono::seconds(1);
+        }
+        else
+        {
+            SetShutdownDelay(shutdownDelay);
+        }
+    }
+    else
+    {
+        SetShutdownDelay(shutdownDelay);
+    }
+}
+
+void ShimOptions::SetShutdownDelay(std::wstring& shutdownDelay)
+{
+    auto str = to_multi_byte_string(shutdownDelay, CP_UTF8);
+    m_fShutdownDelay = std::chrono::milliseconds(std::atoi(str.c_str()));
```

Comment:
Possibly a silly question, but what happens if I set the delay to `"-1"`?

Reply:
It would probably behave the same as 0. Looking at the types, negative time seems fine and the thread sleep converts the duration into absolute time and looks at if the time is before "now". I'll add a check though and throw for values less than 0.